x-pack/plugins/enterprise_search/public/applications/shared/layout/nav.tsx

```diff
@@ -15,11 +15,12 @@ import {
   ENTERPRISE_SEARCH_OVERVIEW_PLUGIN,
   WORKPLACE_SEARCH_PLUGIN,
 } from '../../../../common/constants';
+import { ProductAccess } from '../../../../common/types';
 import { SEARCH_INDICES_PATH } from '../../enterprise_search_content/routes';
 
 import { generateNavLink } from './nav_link_helpers';
 
-export const useEnterpriseSearchNav = () => {
+export const useEnterpriseSearchNav = (access: ProductAccess) => {
```

Comment:
What if we default this `access` to `NO_ACCESS`? Then we'd only need to import it here.

That way if `internalData.access` is undefined this function will handle the defaulting, instead of needing `internalData.access || NO_ACCESS` in all the callers.
```suggestion
export const useEnterpriseSearchNav = (access?: ProductAccess = NO_ACCESS) => {
```

Reply:
I think if we do that, we run the significant risk of developers forgetting to add the correct access parameter. That is something that should surface in testing, but catching it with types will save us a lot of headaches.

The fact that we're needing to do that still suggests to me that our approach here is too scattered. ProductAccess is a single value across the entire application, we should be able to access it directly.